### PR TITLE
🌱 KCP: use new registry for Kubernetes >= v1.22

### LINKED ISSUE #7472: Use new Kubernetes registry for Kubernetes >= v1.22
Given:
* that the new registry is cheaper for the project and
* there's an upstream effort (kubeadm) to backport the registry change back until v1.22

let's align and already use the new registry for Kubernetes >= v1.22

xrefs:
* https://github.com/kubernetes/kubeadm/issues/2671#issuecomment-1290991218
* https://github.com/kubernetes/kubeadm/issues/2671#issuecomment-1293373083

/kind feature

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Before this PR we started using the new registry with Kubernetes 1.25.

Given:
* that the new registry is cheaper for the project and
* there's an upstream effort (kubeadm) to backport the registry change back until v1.22 

let's align and already use the new registry for Kubernetes >= v1.22

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #7472

### CODE REVIEW

**Review by chrischdi — COMMENTED** [review]
/lgtm

**Review by killianmuldoon — APPROVED** [review]
/lgtm

File: controlplane/kubeadm/internal/workload_cluster.go
Comment (CecileRobertMichon) [inline controlplane/kubeadm/internal/workload_cluster.go:90]: are the images available for all 1.22 patches (ie. not starting at a specific patch)?

Comment (sbueringer) [reply controlplane/kubeadm/internal/workload_cluster.go:90]:
```bash
for i in {0..15}
do
  docker pull registry.k8s.io/kube-apiserver:v1.22.${i}
  docker pull registry.k8s.io/kube-controller-manager:v1.22.${i}
  docker pull registry.k8s.io/kube-scheduler:v1.22.${i}
  docker pull registry.k8s.io/kube-proxy:v1.22.${i}
done
```

Worked for me.

Tried a few etcd images. Seem to be there as well.

@neolit123 Do you have some information about which related images should be available in the new registry? I guess there was some kind of sync effort?

Comment (fabriziopandini) [reply controlplane/kubeadm/internal/workload_cluster.go:90]: might be @ameukam can help here

**Comment by fabriziopandini** [comment]
lgtm pending answer to the comment about image availability for the entire 1.22 series (as well as for newer series)

File: controlplane/kubeadm/internal/workload_cluster.go
Comment (sbueringer) [reply controlplane/kubeadm/internal/workload_cluster.go:90]: Let's wait for an answer. I don't have a good way to verify that all images exist

**Comment by CecileRobertMichon** [comment]
> lgtm pending answer to the comment about image availability for the entire 1.22 series (as well as for newer series)

same ^

File: controlplane/kubeadm/internal/workload_cluster.go
Comment (ameukam) [reply controlplane/kubeadm/internal/workload_cluster.go:90]: All the images present in `k8s.gcr.io` will be available in `registry.k8s.io`. the promotion process covert both endpoints.

@sbueringer you can use `crane` or `imgpkg` to verify this. 

```bash
imgpkg tag list -i registry.k8s.io/kube-proxy | grep '1.22.'                                                                    12:55:05 AM
sha256-b137918392d06ed92c521d93c105b12224604e8e425c1e30ec194d665e1db9a8.sig
v1.22.0
v1.22.0-alpha.0
v1.22.0-alpha.1
v1.22.0-alpha.2
v1.22.0-alpha.3
v1.22.0-beta.0
v1.22.0-beta.1
v1.22.0-beta.2
v1.22.0-rc.0
v1.22.1
v1.22.1-rc.0
v1.22.10
v1.22.10-rc.0
v1.22.11
v1.22.11-rc.0
v1.22.12
v1.22.12-rc.0
v1.22.13
v1.22.13-rc.0
v1.22.14
v1.22.14-rc.0
v1.22.15
v1.22.15-rc.0
v1.22.16-rc.0
v1.22.2
v1.22.2-rc.0
v1.22.3
v1.22.3-rc.0
v1.22.4
v1.22.4-rc.0
v1.22.5
v1.22.5-rc.0
v1.22.6
v1.22.6-rc.0
v1.22.7
v1.22.7-rc.0
v1.22.8
v1.22.8-rc.0
v1.22.9
v1.22.9-rc.0
```

**Comment by ameukam** [comment]
@fabriziopandini @CecileRobertMichon https://github.com/kubernetes-sigs/cluster-api/pull/7471#discussion_r1013495730

File: controlplane/kubeadm/internal/workload_cluster.go
Comment (sbueringer) [reply controlplane/kubeadm/internal/workload_cluster.go:90]: I think our concern was mostly about old images as we assumed the image promotion process was changed a lot later then the 1.22, ... releases and we didn't know if there was some sort of additional sync.

But I assume this
> All the images present in k8s.gcr.io will be available in registry.k8s.io
means there was a full sync from k8s.gcr.io to registry.k8s.io

**Comment by fabriziopandini** [comment]
> All the images present in k8s.gcr.io will be available in registry.k8s.io. the promotion process covert both endpoints.

thanks @ameukam for context!

/lgtm
/approve

**Pull request merged**
